### DISCUSSION TITLE
please add api proxy for gemini

### DISCUSSION THREAD

**Discussion by weiyi88** [langgenius/dify#4729] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

all know gemini can't use in island , Please add api proxy text box to gemini
<img width="740" alt="image" src="https://github.com/langgenius/dify/assets/32037147/b5f7ffe4-2d00-4b59-9bd8-6a9c6da24a45">

### 2. Describe the feature you'd like to see

All models outside the wall have api proxy text boxes

### 3. How will this feature improve your workflow or experience?

it helpful

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by Soabbar** [comment]
Actually not only for Google, but all that requires API key, would be great to have proxy url field to access from Mainland China

**Comment by weiyi88** [comment]
can add this feature in next version 
or how to feature by self

**Comment by Benniepie** [comment]
I have this same problem - Any help in implementing this would be appreciated - this is stopping me from using Dify - I'm in UK, the The Gemini Pro (Free API) is not available 1unless I proxy my connection to another country. I can achieve this with python requests proxy but not sure how I go about doing this in Dify. Any advice on how to do this myself would be appreciated.

**Reply by Benniepie** [reply]
Here's the info:
https://ai.google.dev/pricing?authuser=1
Gemini Pro API Free Tier is not available to users connecting from UK, EEA (inc EU) and CH - however connecting via another country (e.g. USA) allows free API access.

**Comment by zhangzw16** [comment]
maybe you can try this way https://github.com/langgenius/dify/issues/5669#issuecomment-2195052423,  if u are using docker
